tuta-sdk/rust/src/instance_mapper.rs

```diff
@@ -1,481 +1,516 @@
-                ) => {
-                    mapped.insert(association_name, JsonElement::Array(vec![JsonElement::String(id_tuple.list_id), JsonElement::String(id_tuple.element_id)]));
-                }
-                (AssociationType::BlobElementAssociation, _, ElementValue::Array(elements)) => {
-                    // Blobs ate copied as-is for now
-                    let serialized_aggregates = self.make_serialized_aggregated_array(&association_name, &association_type_ref, elements)?;
-                    mapped.insert(association_name, JsonElement::Array(serialized_aggregates));
-                }
-                _ => {}
+    fn deserialize_string<V>(self, visitor: V) -> Result<V::Value, Self::Error> where V: Visitor<'de> {
+        return match self.value {
+            // Currently we just generate Id types as String but we could be more precise, then we would need to implement newtype
+            ElementValue::String(str) | ElementValue::GeneratedId(str) | ElementValue::CustomId(str) => {
+                visitor.visit_string(str)
+            }
+            _ => {
+                Err(de::Error::invalid_type(self.value.as_unexpected(), &"string"))
             }
+        };
+    }
+
+    fn deserialize_option<V>(self, visitor: V) -> Result<V::Value, Self::Error> where V: Visitor<'de> {
+        match self.value {
+            ElementValue::Null => visitor.visit_none(),
+            // We have the same value but the visitor who is driving us will proceed to inner type
+            _ => visitor.visit_some(self)
         }
+    }
 
-        Ok(mapped)
+    fn deserialize_seq<V>(self, visitor: V) -> Result<V::Value, Self::Error> where V: Visitor<'de> {
+        if let ElementValue::Array(arr) = self.value {
+            visitor.visit_seq(SeqDeserializer::new(arr.into_iter()))
+        } else {
+            Err(de::Error::invalid_type(self.value.as_unexpected(), &"sequence"))
+        }
     }
 
-    /// Creates a JSON array from an aggregated array
-    fn make_serialized_aggregated_array(&self, association_name: &String, association_type_ref: &TypeRef, elements: Vec<ElementValue>) -> Result<Vec<JsonElement>, InstanceMapperError> {
-        let mut serialized_elements: Vec<JsonElement> = Vec::new();
-        for element in elements {
-            match element {
-                ElementValue::Dict(a) => {
-                    let serialized = self.serialize(&association_type_ref, a)?;
-                    serialized_elements.push(JsonElement::Dict(serialized));
-                }
-                ElementValue::String(v) => {
-                    serialized_elements.push(JsonElement::String(v));
+    fn deserialize_struct<V>(
+        self,
+        name: &'static str,
+        fields: &'static [&'static str],
+        visitor: V,
+    ) -> Result<V::Value, Self::Error> where V: Visitor<'de> {
+        if name == "IdTuple" {
+            struct IdTupleMapAccess<I: Iterator<Item=(&'static str, String)>> {
```

Comment:
we can probably nuke it, since we make iter anyway we can use blanket impl for iter